gitpod-io/gitpod
# Add label to differentiate backup/restore total/failure is PVC or GCS

## Description

Add label to differentiate backup/restore total/failure is PVC or GCS

## Related Issue(s)

Fixes #11722

## How to test

1. Launch workspace and write some file
2. Stop workspace and reopen workspace
3. Check the ws-manager metric `curl -XGET localhost:9500/metrics`
4. Verify that the following metrics with the label `pvc` is `false` and the counter increase
 - gitpod_ws_manager_workspace_backups_failure_total
 - gitpod_ws_manager_workspace_backups_total
 - gitpod_ws_manager_workspace_restores_failure_total
 - gitpod_ws_manager_workspace_restores_total
5. Enable feature flag PVC
6. Rerun steps 1-3
7. Verify that the following metrics with the label `pvc` is `true` and the counter increase
 - gitpod_ws_manager_workspace_backups_failure_total
 - gitpod_ws_manager_workspace_backups_total
 - gitpod_ws_manager_workspace_restores_failure_total
 - gitpod_ws_manager_workspace_restores_total

After that, we could use the label `pvc` is `false` _or_ `true` to differentiate whether the backup/restore is by GCS or PVC.

Examples:
<img width="922" alt="image" src="https://user-images.githubusercontent.com/49380831/181512545-e71e765f-1e5e-4555-8efb-ecdfcd712294.png">
<img width="941" alt="image" src="https://user-images.githubusercontent.com/49380831/181512681-04bfb43a-6daf-4d7d-9555-8ce22107b546.png">

## Release Notes

```release-note
None
```

## Documentation

None

## Werft options:
